scene/gui/item_list.cpp

```diff
@@ -756,6 +756,17 @@ void ItemList::gui_input(const Ref<InputEvent> &p_event) {
 					}
 				}
 
+				// Because the above signal handler might have modified the list of Items, we need to
+				// check if we are empty now, in which case we should not emit a signal with a now
+				// faulty index.
+				if (items.is_empty()) {
+					return;
+				}
```

Comment:
This check could be moved to `get_item_at_position()` and return -1.
Then you could return if `i` is -1.